File: go/sqrt-x/sqrt-x.go
package main

import(
	"fmt"

)

func mySqrt(x int) int {
	root := x
	for root*root > x{
		root = (root + x / root) / 2 
	}
	return root
}

func main(){
	x := 8
	fmt.Println("answer to the first example is:",mySqrt(x))
}

File: js/sqrt-x/sqrt-x.js
function mySqrt(x){
    if (x < 2) return x
    let root = x / 2
    while (true) {
        const approximateRoot = (root + x / root) / 2
        if (Math.floor(approximateRoot) === Math.floor(root)) {
            return Math.floor(approximateRoot)
        }
        root = approximateRoot
    }
}
let x=4
console.log("answer to the first example is:",mySqrt(4))